### DISCUSSION TITLE
why so many plugins shows "not available in code-server for web"? and how can I solve it?

### DISCUSSION THREAD

**Discussion by SimonMing47** [coder/code-server#6406] [Q&A]
`background`: **Most of** the plugins are available, I don't know how to solve it.

`details`:
like this picture says
![Screenshot_2023-08-29-13-42-49-905_com.microsoft.emmx-edit.jpg](https://s2.loli.net/2023/08/29/pWiLXT7IQDVzNBU.jpg)

in addition, the picture below is the version of code-server I use:
![Screenshot_2023-08-29-18-45-30-100_com.termux-edit.jpg](https://s2.loli.net/2023/08/29/nZC2idvzuG8tyFI.jpg)

I install it in termux in Android os
I've already use `pkg upgrade` command to upgrade the code-server.

**Comment by code-asher** [comment]
I am not sure...these extensions work for me on Ubuntu so maybe it is a platform issue.  Is there any difference if you install via the command line? 

```
code-server --install-extension golang.go
```

**Reply by SimonMing47** [reply]
nope

![Screenshot_2023-08-30-08-10-16-053_com.termux-edit.jpg](https://s2.loli.net/2023/08/30/48EQDloyqngGTOj.jpg)

**Reply by code-asher** [reply]
I did some digging and it looks like on Android `process.platform` returns `android` but VS Code does not handle that, so it defaults to `web` instead.

https://github.com/microsoft/vscode/blob/fe0237c03844f97191fc038ecf963ddaa2b39c51/src/vs/base/common/platform.ts#L154-L161

https://github.com/microsoft/vscode/blob/fe0237c03844f97191fc038ecf963ddaa2b39c51/src/vs/platform/extensionManagement/common/extensionManagementUtil.ts#L191-L196

https://github.com/microsoft/vscode/blob/fe0237c03844f97191fc038ecf963ddaa2b39c51/src/vs/platform/extensionManagement/common/extensionManagement.ts#L70-L116

**Reply by code-asher** [reply]
Looks like there might be a few more places where `android` would need to be handled, so if someone tries fixing this upstream then all of those should be addressed as well.

**Reply by SteveWorkshop** [reply]
> Looks like there might be a few more places where `android` would need to be handled, so if someone tries fixing this upstream then all of those should be addressed as well.

try reporting it to ms？
